### ISSUE TITLE
Remove resource hints when sanitizing scripts

### ISSUE THREAD

**Issue by swissspidy** [ampproject/amp-wp#2334]
I just had a plugin add some JavaScript for Google Tag Manager. The AMP plugin correctly removes the custom JavaScript, but there's still the `<link rel="dns-prefetch" href="//www.googletagmanager.com">` tag on the page.

The same goes for other resource hints like `<link rel="dns-prefetch" href="//s.w.org">`.

**Comment by westonruter** [comment]
Do you have the sample plugin code that originally added the GTM script?

I'm not sure that it would be worth it to try to remove these resource hints automatically. They aren't invalid AMP, and it could be very difficult to determine if the resource hint is actually desired or not.

In the end, the AMP validation error is sanitized and the resource hint is not invalid AMP. Sure it's not great to leave a stray resource hint that is unused, but that's really something that a developer should fix while they fix the underlying problem with  the original script.

On the other hand, the pre-connect for `//s.w.org` is probably just erroneously not being removed as part of this logic:

https://github.com/ampproject/amp-wp/blob/9c49871bd4c961e319983154cfee4f755201c35d/includes/class-amp-theme-support.php#L859-L861

We probably just need to add a `wp_resource_hints` filter to remove it, no?

**Comment by swissspidy** [comment]
> Do you have the sample plugin code that originally added the GTM script?

It's called Site Kit 🙃

**Comment by westonruter** [comment]
I suggest contacting the developer of that plugin to add AMP compatibility. 😉

**Comment by westonruter** [comment]
I was about to add this code:

```php
add_filter(
	'wp_resource_hints',
	function ( $urls, $relation_type ) {
		if ( 'dns-prefetch' === $relation_type ) {
			$urls = array_diff(
				$urls,
				array(
					/** This filter is documented in wp-includes/formatting.php */
					apply_filters( 'emoji_svg_url', 'https://s.w.org/images/core/emoji/12.0.0-1/svg/' ),
				)
			);
		}
		return $urls;
	},
	10,
	2
);
```

But then I discovered the function `wp_staticize_emoji()`. Would it not be better to add that on `the_content`, `the_title`, etc? This would largely fix #204, as the logic would not have to be added to a sanitizer as suggested in https://github.com/ampproject/amp-wp/issues/204#issuecomment-470338101.

**Comment by westonruter** [comment]
> But then I discovered the function `wp_staticize_emoji()`. Would it not be better to add that on `the_content`, `the_title`, etc? This would largely fix #204, as the logic would not have to be added to a sanitizer as suggested in [#204 (comment)](https://github.com/ampproject/amp-wp/issues/204#issuecomment-470338101).

Implemented in #2348.
